db/cache/charts.go

```diff
@@ -1117,117 +1239,283 @@ func hashrate(time, chainwork ChartUints) (ChartUints, ChartUints) {
 	return t, y
 }
 
+// dailyHashrate provides the provided daily chainwork data to hashrate data.
```

Comment:
Seems like "converts" would be clearer than "provides".  Or "computes hashrate data from daily chainwork data".